### ISSUE TITLE
android.content.res.Resources$NotFoundException with 0.14.+

### ISSUE THREAD

**Issue by rockerhieu** [robolectric/robolectric-gradle-plugin#110]
After upgrading to 0.14.+ (which is required by Android Studio 1.0RC1), all test cases are failed because of this issue:

```
java.lang.RuntimeException: android.content.res.Resources$NotFoundException: unknown resource 2131624131
	at org.robolectric.RobolectricTestRunner$2.evaluate(RobolectricTestRunner.java:226)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.robolectric.RobolectricTestRunner$1.evaluate(RobolectricTestRunner.java:158)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecuter.runTestClass(JUnitTestClassExecuter.java:86)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecuter.execute(JUnitTestClassExecuter.java:49)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassProcessor.processTestClass(JUnitTestClassProcessor.java:69)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:48)
	at sun.reflect.GeneratedMethodAccessor66.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.messaging.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:32)
	at org.gradle.messaging.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93)
	at com.sun.proxy.$Proxy2.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass(TestWorker.java:105)
	at sun.reflect.GeneratedMethodAccessor65.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.messaging.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:355)
	at org.gradle.internal.concurrent.DefaultExecutorFactory$StoppableExecutorImpl$1.run(DefaultExecutorFactory.java:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
```

I already tried:

* `@Config(emulateSdk = 18, manifest = "./src/main/AndroidManifest.xml")`
* `@Config(emulateSdk = 18, manifest = "./app/src/main/AndroidManifest.xml")`
* `@Config(emulateSdk = 18, manifest = "src/main/AndroidManifest.xml")`
* `@Config(emulateSdk = 18, manifest = "app/src/main/AndroidManifest.xml")`

All of the above configurations didn't help. The issue is persisted.

Do you have any idea to fix this issue? Thanks!

**Comment by RomanZhilich** [comment]
I've got same issue, the only thing that helped is removing ```applicationId``` from build.gradle
And I don't like this solution at all, I actually thought ```applicationId``` is mandatory.

**Comment by venkatraman7** [comment]
Hi Roman,
I resolved my issue by updating Robolectric to 2.4 and android support to
com.android.support:support-v13:19.1.0 and SDK version to 19. In addition
to this updated the ordertype values in iml files also fixes this issue. No
make over needed other than this.

Thanks
Venkat

**Comment by RomanZhilich** [comment]
Nope, that doesn't seem to help
I've already been using Robolectric 2.4, and I don't use support library.
I've set SDK version to 19 and used [this](https://github.com/robolectric/robolectric-gradle-plugin/wiki/Gradle-Task-to-Fix-IML-File) to reorder iml entries, but exception is still there.

**Comment by erd** [comment]
The issue with `android.content.res.Resources$NotFoundException: unknown resource 2131624131` should be fixed with #129 since we rely on the Android Gradle plugin to merge resources from libraries and into the main app before running tests.

**Comment by ligi** [comment]
It seems not completely fixed with #129 - works in AndroidStudio for me but still get a android.content.res.Resources$NotFoundException when running from command-line https://github.com/ligi/AndroidNewUnitTestTest/tree/with_robolectric
